include/zephyr/sys/util.h

```diff
@@ -578,6 +578,15 @@ char *utf8_lcpy(char *dst, const char *src, size_t n);
 /** @brief Number of Hz in @p x MHz */
 #define MHZ(x) (KHZ(x) * 1000)
 
+#if defined(CONFIG_ARCH_POSIX)
+#define __z_POSIX_ARCH_WAIT_A_BIT k_busy_wait(10)
+/* In the POSIX ARCH, code takes zero time to execute,
+ * so busy wait loops become infinite loops, unless we
+ * force the loop to take a bit of time
+ */
+#else
+#define __z_POSIX_ARCH_WAIT_A_BIT
+#endif
```

Comment:
If this is an API in a general header, can we name it something generic?  "_Z_SPIN_DELAY" or something, with a comment explaining the requirement on POSIX that it be present in any spin loop?

Reply:
I did not mean it as an API, but as a hidden macro for WAIT_FOR below. We can add such a general purpose macro as you mention though.

Reply:
Ok, any preferred name?

Reply:
Since we are talking about naming here, please do not use underscores since they are "reserved."

Reply:
Z_SPIN_DELAY()? Otherwise just give a name  :)

Reply:
Your code, your call.  I think Z_SPIN_DELAY is great personally, but I'm not picky (even about underscores, heh).